stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -123,6 +123,7 @@ factorize(D::Diagonal) = D
 real(D::Diagonal) = Diagonal(real(D.diag))
 imag(D::Diagonal) = Diagonal(imag(D.diag))
 
+isdiag(D::Diagonal) = true
```

Comment:
Sorry, one more thing. This fails when we have a block diagonal matrix in which all blocks are not diagonal.

MWE:
```
julia> D=Diagonal([[1 0; 0 1], [1 0; 1 1]])
2×2 Diagonal{Array{Int64,2},Array{Array{Int64,2},1}}:
 [1 0; 0 1]      ⋅     
     ⋅       [1 0; 1 1]

julia> isdiag(D)
true
```

I believe it should be
`isdiag(D::Diagonal) = all(isdiag, D.diag)`
`isdiag(D::Diagonal{<:Number}) = true`